hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/CloneSnapshotProcedure.java

```diff
@@ -457,6 +457,10 @@ private void postCloneSnapshot(final MasterProcedureEnv env)
     // 1. Create Table Descriptor
     // using a copy of descriptor, table will be created enabling first
     final Path tempTableDir = FSUtils.getTableDir(tempdir, tableDescriptor.getTableName());
+    if (FSUtils.isExists(mfs.getFileSystem(), tempTableDir)) {
+      LOG.warn("temp table dir already exists on disk: {}, will be deleted.", tempTableDir);
+      FSUtils.deleteDirectory(mfs.getFileSystem(), tempTableDir);
+    }
```

Comment:
The problem is: the retry will never succeed if the region dirs exist? Add comment here?

Reply:
yes, RestoreSnapshotHelper.restoreHdfsRegions will remove the region dirs, but CloneSnapshotProcedure.createFilesystemLayout not allow exist remove. retrying will have the same effect

Reply:
Sorry. I didn't follow here is why region dir changed? Add more comments to code too?

Reply:
if ok, I will update other commit